inc/saimacsec.h

```diff
@@ -758,18 +758,27 @@ typedef enum _sai_macsec_sa_attr_t
      * @default 0
      * @validonly SAI_MACSEC_SA_ATTR_MACSEC_DIRECTION == SAI_MACSEC_DIRECTION_EGRESS
      */
-    SAI_MACSEC_SA_ATTR_XPN,
+    SAI_MACSEC_SA_ATTR_INITIAL_EGRESS_XPN,
 
     /**
-     * @brief Minimum value of ingress MACsec packet number (PN/XPN).
+     * @brief MACsec current packet number (PN/XPN). For ingress, largest
+     * received packet number. For egress, 1 less than the next packet number.
+     *
+     * @type sai_uint64_t
+     * @flags READ_ONLY
+     */
+    SAI_MACSEC_SA_ATTR_CURRENT_XPN,
+
+    /**
+     * @brief Configured minimum acceptable ingress MACsec packet number (PN/XPN).
      * Updated by value from MACsec peer by Key Agreement protocol.
      *
      * @type sai_uint64_t
      * @flags CREATE_AND_SET
      * @default 1
      * @validonly SAI_MACSEC_SA_ATTR_MACSEC_DIRECTION == SAI_MACSEC_DIRECTION_INGRESS
      */
-    SAI_MACSEC_SA_ATTR_MINIMUM_XPN,
+    SAI_MACSEC_SA_ATTR_MINIMUM_INGRESS_XPN,
```

Comment:
i think we can do alias here so that we can maintain backward compatible here.

Reply:
I would rather change the 3 attribute names to:
SAI_MACSEC_SA_ATTR_CONFIGURED_EGRESS_XPN (new attribute)
SAI_MACSEC_SA_ATTR_XPN (changed from CREATE_AND_SET to READ_ONLY) **
SAI_MACSEC_SA_ATTR_MINIMUM_XPN (no change)

** There is no need to change the default value of this attribute for normal MACsec operation.  If existing NOS code never set this attribute, it would work and there would be no backward compatibility issue.

I prefer this approach to attribute name alias,  But the other deterrent is header compile (meta/make) fails with a bunch of parsing warnings when I try ATTR2 = ATTR1, preceded by @ignore flag.

Reply:
What errors you get ?

Reply:
The parse warning & resulting compile failure goes away if we put the ignore in 1 line likes this:
    /** @ignore - for backward compatibility */
I will use this and do another commit.

Actually the parse errors occurred if I did it in the usual style the compile failure was preceded by the following parse error:
WARNING: empty line required between each below elements:                                        
WARNING: desired elemen order: @brief @warning? description? @attributes? @params? @return?      WARNING: invalid spacing (@) on saimacsec.h:                                                     
/**                                                                                                   * @ignore - for backward compatibility                                                      
     */

Reply:
seems like you get wrong spaces, checkout other "@ignore" attributes and try to follow them